packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -114,22 +122,35 @@ class SocketReference extends TypedEventEmitter<ISocketEvents> {
         assert(!SocketReference.socketIoSockets.has(key), 0x220 /* "socket key collision" */);
         SocketReference.socketIoSockets.set(key, this);
 
-        // The server always closes the socket after sending this message
-        // fully remove the socket reference now
-        socket.on("server_disconnect", (socketError: IOdspSocketError) => {
+        const disconnectHandler = (socketError: IOdspSocketError, reason: string, clientId?: string) => {
```

Comment:
Did old version of PUSH supply reason already? If it's recent change, is it propagated through PROD?